tests/mtest/src/main/scala/tests/BaseSuite.scala

```diff
@@ -158,6 +163,15 @@ class BaseSuite extends TestSuite {
     }
   }
 
+  def failingAsyncTest(name: String, maxDuration: Duration = Duration("10min"))(
```

Comment:
I think we can avoid some duplicate logic by encoding the failure bit where we collect the tests